src/pybind/mgr/orchestrator.py

```diff
@@ -716,6 +718,69 @@ def __init__(self):
         # some replicaset special sauce for autoscaling?
         self.extended = {}
 
+        # Object with specific settings for the service
+        self.service_spec = None
+
+class RGWSpec(object):
```

Comment:
would it make sense to inherent form `StatelessServiceSpec`?

Reply:
+1

Reply:
I have tried to fit the implementation on what was done previously. In this case the attribute "service_spec" with definition "Object with specific settings for the service" seemed to be thought for store extra settings for each of the different stateless services.

Nothing against to use another design. As is suggested we can have three different "StatelessServiceSpec" child classes for each service (RGWSpec, NFSSpec, MDSSpec).

Ernesto suggested to use the adaptor design pattern to allow the use of specific attributes in each orchestrator. This will imply the "selection" of the right adaptor in orchestrator clients. For example, in the orchestrator_cli._rgw_add(...) we will need to select the right adaptor depending if we are using ansible or rook for example...

I think that in this case what we can do is just to define all the attributes needed in each of the individual stateless services spec classes, and each orchestrator can decide "to/not to/how to" use each attribute.

This will force to make specific checks of the attributes in each orchestrator (i know...not nice), but nevertheless it allows to orchestrator clients to avoid to know what is the backend used.

I nothing against i will create a RGWSpec(StatelessServiceSpec) class.

Reply:
Perhaps it's because I'm more inclined to use classes rather than methods, but my proposal would be:
- Make `StatelessServiceSpec` an abstract class (via `abc` module), so it cannot be used directly. It contains all attributes deemed common for any Stateless service (e.g.: `hosts`) and methods for dealing with it.
  - Create new per-service Spec subclasses by deriving from `StatelessServiceSpec` (e.g.: `RGWServiceSpec`). This class defines all attributes specific to RGW configuration (and probably the existing `StatelessServiceSpec.extended` would be perfect to store this data?).
- On the other hand, create `ServiceAdapter` abstract class (probably empty for now).
  - Derive `AnsibleServiceAdapter` from the above, providing methods to couple any `StatelessServiceSpec` to Ansible requirements (e.g.: if a collection of `extravars` is needed for running a playbook, then `AnsibleServiceAdapter` should provide a `.extravars()` method/property, which basically gets `StatelessService.extended`.

Reply:
I'd like to merge this PR without this change. I'm starting to think that `StatelessServiceSpec` is not the right abstraction. Instead every service has independent needs and trying to get all under one hood doesn't really provide any benefits. I'd like to refactor this to a new structure. But that's out of the scope for this PR

Reply:
will be fixed by #29063